diff --git a/node_modules/react-native-video/ios/Video/Features/RCTPlayerObserver.swift b/node_modules/react-native-video/ios/Video/Features/RCTPlayerObserver.swift
index 339932f..e25cf6a 100644
--- a/node_modules/react-native-video/ios/Video/Features/RCTPlayerObserver.swift
+++ b/node_modules/react-native-video/ios/Video/Features/RCTPlayerObserver.swift
@@ -35,6 +35,7 @@ protocol RCTPlayerObserverHandler: RCTPlayerObserverHandlerObjc {
     func handleDidEnterFullScreen()
     func handleWillExitFullScreen()
     func handleDidExitFullScreen()
+    func handleControlsVisibilityChange(visible: Bool)
 }
 
 // MARK: - RCTPlayerObserver
@@ -357,6 +358,16 @@ class RCTPlayerObserver: NSObject, AVPlayerItemMetadataOutputPushDelegate, AVPla
         }
     #endif
 
+    #if os(tvOS)
+        func playerViewController(
+            _: AVPlayerViewController,
+            willTransitionToVisibilityOfTransportBar visible: Bool,
+            with _: UIViewControllerTransitionCoordinator
+        ) {
+            self._handlers?.handleControlsVisibilityChange(visible: visible)
+        }
+    #endif
+
     func setRestoreUserInterfaceForPIPStopCompletionHandler(_ restore: Bool) {
         guard let _restoreUserInterfaceForPIPStopCompletionHandler else { return }
 
diff --git a/node_modules/react-native-video/ios/Video/RCTVideo.swift b/node_modules/react-native-video/ios/Video/RCTVideo.swift
index 5f2f5f6..23633cb 100644
--- a/node_modules/react-native-video/ios/Video/RCTVideo.swift
+++ b/node_modules/react-native-video/ios/Video/RCTVideo.swift
@@ -132,6 +132,7 @@ class RCTVideo: UIView, RCTVideoPlayerViewControllerDelegate, RCTPlayerObserverH
     @objc var onTextTracks: RCTDirectEventBlock?
     @objc var onAudioTracks: RCTDirectEventBlock?
     @objc var onTextTrackDataChanged: RCTDirectEventBlock?
+    @objc var onControlsVisibilityChange: RCTDirectEventBlock?
 
     @objc
     func _onPictureInPictureEnter() {
@@ -1742,6 +1743,10 @@ class RCTVideo: UIView, RCTVideoPlayerViewControllerDelegate, RCTPlayerObserverH
         self.onVideoFullscreenPlayerDidDismiss?(["target": self.reactTag as Any])
     }
 
+    func handleControlsVisibilityChange(visible: Bool) {
+        self.onControlsVisibilityChange?(["isVisible": visible])
+    }
+
     @objc
     func handleDidFailToFinishPlaying(notification: NSNotification!) {
         guard onVideoError != nil else { return }
diff --git a/node_modules/react-native-video/ios/Video/RCTVideoManager.m b/node_modules/react-native-video/ios/Video/RCTVideoManager.m
index 7ec8955..c6b0cdf 100644
--- a/node_modules/react-native-video/ios/Video/RCTVideoManager.m
+++ b/node_modules/react-native-video/ios/Video/RCTVideoManager.m
@@ -67,6 +67,7 @@ @interface RCT_EXTERN_MODULE (RCTVideoManager, RCTViewManager)
 RCT_EXPORT_VIEW_PROPERTY(onTextTracks, RCTDirectEventBlock);
 RCT_EXPORT_VIEW_PROPERTY(onAudioTracks, RCTDirectEventBlock);
 RCT_EXPORT_VIEW_PROPERTY(onTextTrackDataChanged, RCTDirectEventBlock);
+RCT_EXPORT_VIEW_PROPERTY(onControlsVisibilityChange, RCTDirectEventBlock);
 
 RCT_EXTERN_METHOD(seekCmd : (nonnull NSNumber*)reactTag time : (nonnull NSNumber*)time tolerance : (nonnull NSNumber*)tolerance)
 RCT_EXTERN_METHOD(setLicenseResultCmd : (nonnull NSNumber*)reactTag license : (NSString*)license licenseUrl : (NSString*)licenseUrl)
diff --git a/node_modules/react-native-video/src/Video.tsx b/node_modules/react-native-video/src/Video.tsx
index 789d915..6af1d59 100644
--- a/node_modules/react-native-video/src/Video.tsx
+++ b/node_modules/react-native-video/src/Video.tsx
@@ -219,7 +219,7 @@ const Video = forwardRef<VideoRef, ReactVideoProps>(
         if (!uri && _source.ad?.type !== 'ssai') {
           console.log('Trying to load empty source');
         }
-        const isNetwork = !!(uri && uri.match(/^(rtp|rtsp|http|https):/));
+        const isNetwork = !!(uri && uri.match(/^(rtp|rtsp|http|https|jellyfin-multi):/));
         const isAsset = !!(
           uri &&
           uri.match(
